obsreport/obsreporttest/obsreporttest.go

```diff
@@ -102,6 +115,30 @@ func SetupTelemetry() (TestTelemetry, error) {
 	return settings, err
 }
 
+// SetupOpenTelemetry does setup the testing environment to check the metrics produced by the OpenTelemetry Go SDk,
+// and recorded by receivers, producers or exporters.
+// The caller should defer a call to Shutdown the returned TestTelemetry.
+func SetupOpenTelemetry() (TestTelemetry, error) {
```

Comment:
Can you embed this into the `SetupTelemetry`?

Reply:
I'm not sure if we can, we need a way for the user of `TestTelemetry` to decide if it should use the OC views checks or OTel prometheus check.

If we embed it to `SetupTelemetry` we would need to add a `TestTelemetry.CheckWithOtel()` method for changing the underlying check, or something like that.

Do you have other suggestion for this?

Reply:
Use the `featuregate` that we have for checking that?

Reply:
Sure, but we can't just pass a `featuregate.Registry` to `SetupTelemetry()` because that would be a breaking change, right? Or don't we care about this on the `obsreporttest` package?

Are we okay by just adding a `featuregate.Registry` as a field of the `TestTelemtry`, so we would be using like:
```go
tt, err := SetupTelemetry()
// handle err

// change the TestTelemetry registry
tt.Registry = registry

// use tt to check things based on its registry?
```

Reply:
Can we simply change the entire package (for receiver) to check (for the moment) metrics via prometheus since opencensus also reports there? Then we revert that and check via a TestExporter?

> Are we okay by just adding a featuregate.Registry as a field of the TestTelemtry, so we would be using like:

I think we should better expose a boolean "OtelChecker" and will remove later.

Reply:
> Can we simply change the entire package (for receiver) to check (for the moment) metrics via prometheus since opencensus also reports there? 

I didn't want to touch the OC tests at first. But since we're ok with it, we can get rid of this separation oc x otel, and create a prometheus handler with both sdks to do assertions.